tests/DataBuilder/WebsiteDataReaderTest.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Website\Tests\DataBuilder;
+
+use Doctrine\Website\DataBuilder\WebsiteData;
+use Doctrine\Website\DataBuilder\WebsiteDataReader;
+use org\bovigo\vfs\vfsStream;
+use PHPUnit\Framework\TestCase;
+use RuntimeException;
+
+class WebsiteDataReaderTest extends TestCase
+{
+    private WebsiteDataReader $websiteDataReader;
+
+    public function testRead(): void
+    {
+        $websiteData = $this->websiteDataReader->read('valid');
+
+        $expected = new WebsiteData('valid', []);
+
+        self::assertEquals($expected, $websiteData);
+    }
+
+    public function testReadInvalidFilePath(): void
+    {
+        $this->expectException(RuntimeException::class);
+        $this->expectExceptionMessage('File vfs://cache/data/foo.json does not exist. Run ./doctrine build-website-data to generate.');
+
+        $this->websiteDataReader->read('foo');
+    }
+
+    public function testReadInvalidJson(): void
+    {
+        $this->expectException(RuntimeException::class);
+        $this->expectExceptionMessage('Could not load JSON from file vfs://cache/data/invalid.json');
+
+        $this->websiteDataReader->read('invalid');
+    }
+
+    /**
+     * A PHPUnit warning is expected here because of file_get_contents not being able to open a file.
```

Comment:
When I run phpunit, it tells me that this is actually a PHP warning:

```
1 test triggered 1 PHP warning:

1) /home/gregoire/dev/doctrine-website/lib/DataBuilder/WebsiteDataReader.php:33
file_get_contents(vfs://cache/data/im-a-directory-stupid.json): Failed to open stream: "org\bovigo\vfs\vfsStreamWrapper::stream_open" call failed

Triggered by:

* Doctrine\Website\Tests\DataBuilder\WebsiteDataReaderTest::testReadUnableToReadFileContent
  /home/gregoire/dev/doctrine-website/tests/DataBuilder/WebsiteDataReaderTest.php:45

```

Reply:
It's vfsStream that doesn't like it.  I've removed this test for now because it's not an important one. Maybe I'll use the real filesystem like in other tests for this case.